packages/playwright-test/src/plugins/webServerPlugin.ts

```diff
@@ -92,47 +96,48 @@ export class WebServerPlugin implements TestRunnerPlugin {
     }
 
     debugWebServer(`Starting WebServer process ${this._options.command}...`);
-    const { launchedProcess, kill } = await launchProcess({
-      command: this._options.command,
+
+    this._childProcess = spawn(this._options.command, [], {
       env: {
         ...DEFAULT_ENVIRONMENT_VARIABLES,
         ...envWithoutExperimentalLoaderOptions(),
         ...this._options.env,
       },
       cwd: this._options.cwd,
-      stdio: 'stdin',
+      stdio: 'pipe',
       shell: true,
-      attemptToGracefullyClose: async () => {},
-      log: () => {},
-      onExit: code => processExitedReject(new Error(code ? `Process from config.webServer was not able to start. Exit code: ${code}` : 'Process from config.webServer exited early.')),
-      tempDirectories: [],
     });
-    this._killProcess = kill;
+    this._processExitedPromise = new Promise((resolve, reject) => {
+      this._childProcess!.once('exit', code => {
+        if (code)
+          reject(new Error(`Process from config.webServer was not able to start. Exit code: ${code}`));
```

Comment:
I [believe](https://nodejs.org/api/child_process.html#event-exit) the exit code will be `null` when the process was killed by a signal. Should we check for both here?